src/Mod/Draft/WorkingPlane.py

```diff
@@ -1256,11 +1299,15 @@ def align_to_selection(self, offset=0, _hist_add=True):
                 objs.append(Part.getShape(sel.Object, sub, needSubElement=True, retType=1))
 
         if len(objs) != 1:
+            ret = False
             if all([obj[0].isNull() is False and obj[0].ShapeType in ["Edge", "Vertex"] for obj in objs]):
                 ret = self.align_to_edges_vertexes([obj[0] for obj in objs], offset, _hist_add)
-            else:
-                ret = False
-
+            elif all([obj[0].isNull() is False and obj[0].ShapeType in ["Edge", "Face"] for obj in objs]):
+                    edges = [obj[0] for obj in objs if obj[0].ShapeType == "Edge"]
+                    faces = [obj[0] for obj in objs if obj[0].ShapeType == "Face"]
+                    if len(edges) == 1 and len(faces) == 1:
+                        if edges[0].hashCode() in [e.hashCode() for e in faces[0].Edges]:
+                            ret = self.align_to_face_and_edge(faces[0], edges[0], offset)
```

Comment:
The hashCode check should be removed. It does not work if the face and edge are nested in a Std_Part (in which case the shapes are translated separately). Removing this check makes the code more flexible as well. You can select any edge for the X axis direction. The code can even handle the case where the selected edge is parallel to the face normal.

The _hist_add argument should be added for consistency:
`ret = self.align_to_face_and_edge(faces[0], edges[0], offset, _hist_add)`